### PR TITLE
hr_worked_days_from_timesheet migrate and re-factored to Odoo 8.0

### PR DESCRIPTION
**Pull request by acysos**
Hello,

I have migrate and re-factored this module. Less code and improved for inherit.

Greetings,
Ignacio Ibeas

### CODE REVIEW

**Comment by acysos** [comment]
Hello,

I don't understand the travis fail, in the my test repository, pass all test.
https://travis-ci.org/acysos/test-addons/builds/89748588

Greetings

File: hr_worked_days_from_timesheet/models/hr_payslip.py [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+    def timesheet_mapping(self, timesheet_sheets, payslip, date_from, date_to,
+                          date_format):
+        """This function takes timesheet objects imported from the timesheet
+        module and creates a dict of worked days to be created in the payslip.
+        """
+        # Create one worked days record for each timesheet sheet
+        wd_model = self.env['hr.payslip.worked_days']
+        for ts_sheet in timesheet_sheets:
+            # Get formated date from the timesheet sheet
+            date_from_formated = datetime.strptime(
```
Comment (feketemihai) [inline hr_worked_days_from_timesheet/models/hr_payslip.py]: You could use fields.Datetime.from_string and fields.Date.to_string functions available in new api.

File: hr_worked_days_from_timesheet/__openerp__.py [hunk 2]
```diff
@@ -20,31 +21,22 @@
 
 {
     'name': 'Worked Days From Timesheet',
-    'version': '1.0',
+    'version': '8.0.1.0',
```
Comment (feketemihai) [inline hr_worked_days_from_timesheet/__openerp__.py]: Please change the version number according to https://github.com/OCA/maintainer-tools/blob/master/CONTRIBUTING.md#version-numbers

File: hr_worked_days_from_timesheet/README.rst [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+
+* Not known issues
+
+Bug Tracker
+===========
+
+Bugs are tracked on `GitHub Issues <https://github.com/OCA/hr/issues>`_.
+In case of trouble, please check there if your issue has already been reported.
+If you spotted it first, help us smashing it by providing a detailed and welcomed feedback
+`here <https://github.com/OCA/hr/issues/new?body=module:%20hr_worked_days_hourly_rate%0Aversion:%208.0.1.0%0A%0A**Steps%20to%20reproduce**%0A-%20...%0A%0A**Current%20behavior**%0A%0A**Expected%20behavior**>`_.
```
Comment (feketemihai) [inline hr_worked_days_from_timesheet/README.rst]: Please change the name of the module to the right one, it's the link from the hr_worked_days_hourly_rate and thre version...

**Comment by acysos** [comment]
Hello,

Changes done.

Greetings

File: hr_worked_days_from_timesheet/__openerp__.py [hunk 4]
```diff
@@ -20,31 +21,22 @@
...
         'hr_timesheet_sheet',
     ],
     'data': [
-        'hr_payslip_view.xml'
+        'views/hr_payslip_view.xml'
     ],
     'test': [
         'test/worked_days_from_timesheet_test.yml'
     ],
     'demo': [],
```
Comment (feketemihai) [inline hr_worked_days_from_timesheet/__openerp__.py]: Please remove empty key: demo

File: hr_worked_days_from_timesheet/models/hr_payslip.py [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+            timesheet_sheets = self.env['hr_timesheet_sheet.sheet'].search(
+                ['&', '&', '|', '&',
+                 ('date_from', '<=', date_to), ('date_from', '>=', date_from),
+                 '&', ('date_to', '<=', date_to), ('date_to', '>=', date_from),
+                 ('state', '=', 'done'),
+                 ('employee_id', '=', payslip.employee_id.id)]
+            )
+
+            if not timesheet_sheets:
+                raise Exception(
```
Comment (feketemihai) [inline hr_worked_days_from_timesheet/models/hr_payslip.py]: Could you use exception.Warning from openerp.exception, if we leave just python Exception it's going to throw an Odoo Server Error instead a Warnign to the user...

File: hr_worked_days_from_timesheet/test/worked_days_from_timesheet_test.yml [hunk 6]
```diff
@@ -1,16 +1,18 @@
 -
```
Comment (feketemihai) [inline hr_worked_days_from_timesheet/test/worked_days_from_timesheet_test.yml]: Could you change also the test into the new api...

Comment (ghost) [inline hr_worked_days_from_timesheet/test/worked_days_from_timesheet_test.yml]: use python unittest instead of yml. yml tests are always problematic.

**Comment by andhit-r** [comment]
Any news on this? May i offer help to finish it?

**Comment by acysos** [comment]
Hello,

I need to pass to test to unittest to finish it.

Greetings

**Comment by andhit-r** [comment]
@acysos I have added unittest for this module. I created PR to your branch (https://github.com/acysos/hr/tree/8.0-hr_worked_days_from_timesheet) and OCA/hr ( #201 ) just to test.

There are some errors but i think it's not related to this module

**Comment by andhit-r** [comment]
@acysos i need this module migrated to 8.0 soon and willing to finish your work. May i offer help to finish this? I will create new PR for this module and retain commit history from your work. 

Any thought @feketemihai @dufresnedavid @dreispt @max3903 ?

**Comment by acysos** [comment]
@andhit-r The module is working, I have in one of my clients. I have merge you PR.

Thank you

**Comment by andhit-r** [comment]
Thanks @acysos. Travis and runbot still failing though. It seems cause by test on **partner_firstname** module.  Anyone could give advice to make both test green?

**Comment by feketemihai** [comment]
That are solved in #194 but there are not merged yet...

**Comment by andhit-r** [comment]
@acysos @dufresnedavid @dreispt I have confirmed @feketemihai latest comment. #194 will solve travis issues, and hopefully runbot. Would someone review #194 ?

**Comment by acysos** [comment]
@feketemihai The merge is to the OCA principal branch, so I can't merge it. I can add the changes to this branch or you can send a PR.

**Comment by andhit-r** [comment]
@acysos maybe you can review #194 and give your vote. Then you can rebase If #194 merged.

**Comment by andhit-r** [comment]
Any news on this?

**Comment by feketemihai** [comment]
Closed since work will continue in #218 .

**Pull request closed without merging**